src/classes/UTIL_Describe.cls

```diff
@@ -73,11 +73,42 @@ public class UTIL_Describe {
     private static void setInstance(UTIL_Describe utilDescribe){
         utilDescribeInstance = utilDescribe;
     }
+    
+    /*******************************************************************************************************
+    * @description Map display types to a friendly user-readable label.
+    */
+    private static final Map<String, String> LABELS_BY_DISPLAY_TYPE = new Map<String, String> {
+        'ADDRESS' => 'Address',
+        'BASE64' => 'Base64',
+        'BOOLEAN' => 'Checkbox',
+        'COMBOBOX' => 'Picklist',
+        'CURRENCY' => 'Currency',
+        'DATACATEGORYGROUPREFERENCE' => 'DataCategoryGroupReference',
+        'DATE' => 'Date',
+        'DATETIME' => 'Date/Time',
+        'DOUBLE' => 'Number',
+        'EMAIL' => 'Email',
+        'ENCRYPTEDSTRING' => 'Text (Encrypted)',
+        'ID' => 'Id',
+        'INTEGER' => 'Number',
+        'LONG' => 'Text Area (Long)',
+        'MULTIPICKLIST' => 'Picklist (Multi-Select)',
+        'PERCENT' => 'Percent',
+        'PHONE' => 'Phone',
+        'PICKLIST' => 'Picklist',
+        'REFERENCE' => 'Lookup Relationship',
+        'STRING' => 'Text',
+        'TEXTAREA' => 'Text Area',
+        'TIME' => 'Time',
+        'URL' => 'URL'
+    };
 
     /*******************************************
     * Gets describe maps for a new object
     ********************************************/
     static void fillMapsForObject(string objectName) {
+        objectName = objectName.toLowerCase();
```

Comment:
@ptewson-sfdo Is this needed? I'm almost positive that the global map of object results is case-insensitive (even though Map keys are usually case sensitive). We've been using this code for a really long time without any issues. Seems like if the case was a problem we would have seen this long ago.

Reply:
It was a problem and we were seeing maps populated by both mixed-case and lowercased keys